pkg/providers/docker/docker_test.go

```diff
@@ -327,6 +327,38 @@ func TestProviderGenerateDeploymentFileSuccessUpdateMachineTemplate(t *testing.T
 			wantCPFile: "testdata/valid_deployment_cp_expected.yaml",
 			wantMDFile: "testdata/valid_autoscaler_deployment_md_expected.yaml",
 		},
+		{
+			testName: "valid config with rollout",
```

Comment:
I'll leave it up to you but I would prefer if we tested these things through `ControlPlaneSpec` and `WorkersSpec`

That decouples the tests from the fact that these spec builders return yaml (which they should never have)

I understand you are just following whatever was in here, which is fair. But It would be better if we could start writing new tests following a better pattern.

This applies to all providers

Reply:
That makes sense, swapping over to ControlPlaneSpec and WorkersSpec tests instead for all providers